.github/prompts/test-recording.prompt.md

```diff
@@ -0,0 +1,384 @@
+
+```typescript
+export function envSetupForPlayback(): Record<string, string> {
+  return {
+    // Replace actual values with sanitized versions for playback
+    AZURE_CLIENT_ID: "azure_client_id",
+    AZURE_CLIENT_SECRET: "azure_client_secret",
+    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
+    SUBSCRIPTION_ID: "azure_subscription_id",
+    RESOURCE_GROUP: "myjstest",
+    STORAGE_ACCOUNT_NAME: "fakestorageaccount",
+    // Service-specific endpoints
+    ENDPOINT: "https://myaccount.table.core.windows.net/",
+    // Connection strings (sanitized)
+    AZURE_STORAGE_CONNECTION_STRING:
+      "DefaultEndpointsProtocol=https;AccountName=fakestorageaccount;AccountKey=aaaaa;EndpointSuffix=core.windows.net",
+  };
+}
+```
+
+**Key principles:**
+
+- **Use consistent fake values**: Use the same sanitized values across all tests
+- **Maintain format**: Keep the same format as real values (GUIDs, URLs, connection strings)
+- **Match recording expectations**: Values should match what was sanitized in recordings
+- **Include all test dependencies**: Add any environment variables your tests reference
+
+**Common sanitized values:**
+
+- **GUIDs**: `88888888-8888-8888-8888-888888888888`
+- **Subscription IDs**: `azure_subscription_id`
+- **Resource Groups**: `myjstest`
+- **Client credentials**: `azure_client_id`, `azure_client_secret`
+- **Storage accounts**: `fakestorageaccount`
+
+## Quick Commands
+
+```bash
+# Initialize (first time only)
+npx dev-tool test-proxy init
+
+# Restore recordings (creates _recordings/ symbolic link)
+npx dev-tool test-proxy restore
+
+# Record tests
+TEST_MODE=record rushx test
+
+# Test playback (verify recordings work)
+TEST_MODE=playback rushx test
+
+# Push recordings (ONLY when everything is working correctly)
+npx dev-tool test-proxy push
+
+# Reset if corrupted
+npx dev-tool test-proxy reset
+
+# ⚠️ Do NOT pipe any test commands into `head`, `tail`, or similar utilities. Always run the full test suite for accurate results.
+```
+
+## Recording Workflow
+
+**Recommended sequence:**
+
+1. **Restore existing recordings**: `npx dev-tool test-proxy restore`
+2. **Record new/updated tests**: `TEST_MODE=record rushx test`
+3. **Verify playback works**: `TEST_MODE=playback rushx test`
+4. **Fix any sanitizer issues** (see troubleshooting below)
+5. **Repeat steps 2-4 until both record and playback pass**
+6. **Push recordings once**: `npx dev-tool test-proxy push`
+
+**Important**: Only run `test-proxy push` once at the end when all tests pass in both record and playback modes. Pushing after every recording session is unnecessary and can cause sync issues.
+
+## Inspecting Recordings
+
+Recordings can be inspected under the `_recordings/node` (for node) or `_recordings/browser` (for browser) at the package directory. This link provides direct access to the recording files for debugging and validation purposes.
+
+If the `_recordings/` symbolic link is not present in your package directory, run:
```

Comment:
we noticed the symlink is not reliable; in this particular instance(in @KarishmaGhiya's machine) we observed that it doens't generate symlink, throws and goes ahead to proceed with the next steps.

Reply:
Raised https://github.com/Azure/azure-sdk-for-js/issues/34769 to track this.

The agent also isn't super good at following the symbolic link so it might help to actually have a better way of doing this like we were discussing yesterday (either telling it how to navigate to the .assets folder or even an MCP resource for accessing recordings maybe?)